### PR TITLE
Fix 117

### PR DESCRIPTION
**Pull request by lirona**
first changes around pendingApproval -> submit & amount ->shares

### CODE REVIEW

File: contracts/HATVaults.sol [hunk 1]
```diff
@@ -50,7 +50,7 @@ contract  HATVaults is Governable, HATMaster {
     using SafeMath  for uint256;
     using SafeERC20 for IERC20;
 
-    struct PendingApproval {
+    struct SubmittedClaim {
```
Comment (jellegerbrandy) [inline contracts/HATVaults.sol:53]: what about just calling it `Claim`?

File: contracts/HATVaults.sol [hunk 2]
```diff
@@ -91,7 +91,7 @@ contract  HATVaults is Governable, HATMaster {
     //pid -> amount
     mapping(uint256 => uint256) public governanceHatRewards;
     //pid -> PendingApproval
-    mapping(uint256 => PendingApproval) public pendingApprovals;
+    mapping(uint256 => SubmittedClaim) public submittedClaims;
```
Comment (jellegerbrandy) [inline contracts/HATVaults.sol:94]: perhaps `claims` here instead of `submittedClaims`

File: contracts/HATVaults.sol [hunk 3]
```diff
@@ -112,8 +112,8 @@ contract  HATVaults is Governable, HATMaster {
         _;
     }
 
-    modifier noPendingApproval(uint256 _pid) {
-        require(pendingApprovals[_pid].beneficiary == address(0), "HVE02");
+    modifier noSubmittedClaims(uint256 _pid) {
```
Comment (jellegerbrandy) [inline contracts/HATVaults.sol:115]: lets make it `noActiveClaims`

File: contracts/HATVaults.sol [hunk 4]
```diff
@@ -222,34 +222,34 @@ contract  HATVaults is Governable, HATMaster {
...
-     * Upon a call to this function by the committee the pool withdrawn will be disable
-     * till governance will approve or dismiss this pending approval.
-     * @param _pid pool id
-     * @param _beneficiary the approval claim beneficiary
-     * @param _severity approval claim severity
+     * @notice Called by a committee to submit a claim for a bounty.
+     * The submitted claim needs to be approved or dismissed by the Hats governance.
+     * This function should be called only on a safety period, where withdrawals are disabled.
+     * Upon a call to this function by the committee the pool withdrawals will be disabled
+     * untill the Hats governance will approve or dismiss this claim.
```
Comment (jellegerbrandy) [inline contracts/HATVaults.sol:229]: until

**Review by jellegerbrandy — CHANGES_REQUESTED** [review]
Looks great!

I just left some small remarks

File: contracts/HATVaults.sol (hunk 1 above)
Comment (lirona) [reply contracts/HATVaults.sol:53]: i thought so too but there's already a struct called Claim

File: contracts/HATVaults.sol (hunk 2 above)
Comment (lirona) [reply contracts/HATVaults.sol:94]: again, it's for claims that have been submitted by the committee

File: contracts/HATVaults.sol (hunk 3 above)
Comment (lirona) [reply contracts/HATVaults.sol:115]: i would not add another word into this terminology actually..

File: contracts/HATMaster.sol [hunk 5]
```diff
@@ -143,79 +144,86 @@ contract HATMaster is Governable, ReentrancyGuard {
...
-     * @param _lastPoolUpdate lastPoolUpdate
+     * @dev Calculate rewards for a pool by iterating over the history of totalAllocPoints updates,
+     * and sum up all rewards periods from pool.lastRewardBlock untill current block number.
+     * @param _pid The pool id
+     * @param _fromBlock The block from which to start calculation
+     * @param _lastPoolUpdateIndex index of last PoolUpdate in globalPoolUpdates to calculate for
      * @return reward
      */
-    function calcPoolReward(uint256 _pid, uint256 _from, uint256 _lastPoolUpdate) public view returns(uint256 reward) {
+    function calcPoolReward(uint256 _pid, uint256 _fromBlock, uint256 _lastPoolUpdateIndex) public view returns(uint256 reward) {
```
[the remark this answers is not in the document]
Comment (lirona) [reply contracts/HATMaster.sol:214]: agreed, but i want to go over all the alloc point code again (and refactor) at a later time if that's ok.

[the remark this answers is not in the document]
Comment (jellegerbrandy) [reply contracts/HATMaster.sol:214]: yes, I created an issue https://github.com/hats-finance/hats-contracts/issues/135

**Review by jellegerbrandy — APPROVED** [review]

File: contracts/HATMaster.sol (hunk 5 above)
[the remark this answers is not in the document]
Comment (lirona) [reply contracts/HATMaster.sol:214]: great thanks

File: contracts/HATVaults.sol [hunk 6]
```diff
@@ -93,7 +95,7 @@ contract  HATVaults is Governable, HATMaster {
     //pid -> SubmittedClaim
     mapping(uint256 => SubmittedClaim) public submittedClaims;
     //poolId -> (address -> requestTime)
-    mapping(uint256 => mapping(address => uint256)) public withdrawRequests;
+    mapping(uint256 => mapping(address => uint256)) public WithdrawEnableStartTime;
```
Comment (jellegerbrandy) [inline contracts/HATVaults.sol:98]: linter warns that this should be `withdrawEnableStartTime` (i.e. lower case)

**Review by jellegerbrandy — CHANGES_REQUESTED** [review]
I left one comment

File: contracts/HATVaults.sol (hunk 6 above)
Comment (lirona) [reply contracts/HATVaults.sol:98]: oh right my bad

**Review by jellegerbrandy — APPROVED** [review]

**Pull request merged**